### PR TITLE
docs: update contributing docs

### LINKED ISSUE #2431: Update contributing.md
Update CONTRIBUTING.md to include up-to-date instructions on how to contribute to the project

### PR DESCRIPTION
**Pull request by vacekj**
fixes #2431 
fixes https://github.com/gitcoinco/grants-stack/issues/2461

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,27 +1,37 @@
...
 - Any other context that you feel would help resolve the issue
 
 ## Contributing code
 
-We welcome all code contributions, large or small. Please see [DEV.md](packages/round-manager/docs/DEV.md) for instructions on how to set up the project for development.
+We welcome all code contributions, large or small. Please see [DEV.md](docs/DEV.md) for
+instructions on how to set up the project for development.
 
-Before submitting a PR, please make sure that all tests, linting and formatting rules are passing.
+Before submitting a PR, please make sure that all the CI checks are passing. We also utilize git hooks for pre-commit
```
Comment (bard) [inline CONTRIBUTING.md:29]: In the interest of keeping instructions close to the place where they're needed, should this be part of the PR checklist?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,27 +1,37 @@
...
+formatting and pre-push checks, which should help you catch issues early.
 
-We squash all PRs by default. You can request your PR commits to not be squashed if they're well named and organized, and it would make sense to preserve their context.
+We squash all PRs by default. When updating your PR branch, use merge if it's in review, or rebase otherwise, at your
+own discretion.
 
-_To keep the repository clean, we may mark PRs as stale if there has been no activity for more than a week, which are then subject to closure and deletion. If you are stuck, feel free to reach out for help in [Discord](https://discord.gg/gitcoin)._ 
+_To keep the repository clean, we may mark PRs as stale if there has been no activity for more than a week, which are
```
Comment (bard) [inline CONTRIBUTING.md:35]: Good idea!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,27 +1,37 @@
...
+then subject to closure and deletion. If you are stuck, feel free to reach out for help
+in [Discord](https://discord.gg/gitcoin)._
```
Comment (bard) [inline CONTRIBUTING.md:37]: Same considerations as above regarding channel.

File: README.md [hunk 4]
```diff
@@ -8,91 +8,31 @@ This repository contains packages needed for a user to
 Project Maintained via  : `turborepo`
 Package manager         : `pnpm`
 
-## Directory Structure
+## Packages
+
+- round-manager: react-frontend for round-manager
```
Comment (bard) [inline README.md:13]: A native speaker might offer more certainty but I think "React frontend" is the usual spelling.

But that aside, calling it a "frontend for round-manager" might make one think that there's a "backend for round-manager" somewhere, whereas round-manager is the app itself.

File: docs/DEV.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    <Route path="/program/create" element={<CreateProgram />} />
+    <Route path="/program/:id" element={<ViewProgram />} />
+</Route>
+```
+
+Find more information about routing [here](https://reactrouter.com/docs/en/v6).
+
+## Submitting for review
+
+We have git hook
```
Comment (bard) [inline docs/DEV.md:77]: Looks like this was left in the middle?

**Review by bard — APPROVED** [review]

**Review by bhargavaparoksham — APPROVED** [review]

**Review by bard — APPROVED** [review]

**Review by gravityblast — APPROVED** [review]

**Pull request merged**